pkg/controller/azvolume.go

```diff
@@ -153,15 +165,21 @@ func (r *reconcileAzVolume) UpdateStatus(ctx context.Context, volumeName string,
 
 	if isDeleted {
 		if err := r.DeleteFinalizer(ctx, volumeName); err != nil {
-			klog.Errorf("failed to delete finalizer %s for azVolume %s: %v", AzVolumeFinalizer, azVolume.Name, err)
+			klog.Errorf("failed to delete finalizer %s for azVolume %s: %v", azureutils.AzVolumeFinalizer, azVolume.Name, err)
 			return err
 		}
 		return nil
 	}
 
 	updated := azVolume.DeepCopy()
 	if updated.Status == nil {
-		updated.Status = &v1alpha1.AzVolumeStatus{}
+		updated.Status = &v1alpha1.AzVolumeStatus{
+			ResponseObject: status,
+		}
+	} else {
+		// Updating status after update operation
+		updated.Status.ResponseObject.CapacityBytes = status.CapacityBytes
```

Comment:
Is there ever the chance that UpdateStatus will be called with nil status from TriggerUpdate? Might give an error here

Reply:
Ideally, UpdateStatus should be called with nil as status while deletion and it wouldn't enter this code path. But, I have added a check.